Fixes for mobile datepicker

oliverguenther opened the pull request:
- [x] Fix closing the datepicker after "go-next" on duration field on mobile devices due to a wrong button type on cancel button
- [x] Treat mobile datepicker modal as other modals, being more or less fullscreen with a black background (and close button for the scope of this PR)
- [x] Add CDK focus trapping on the datepicker for good measure
- [x] The start date, finish date and duration fields open the numeric keyboard, not alpha-numeric one.

https://community.openproject.org/projects/openproject/work_packages/44144/activity?query_id=3510

oliverguenther committed:
Fix button types so that mobile doesn't use cancel as submit

oliverguenther committed:
Add cdk focus trapping

oliverguenther committed:
Fix vertical align of datepicker month and year select

oliverguenther committed:
Show modal with background and close on mobile

oliverguenther committed:
Align flatpickr-container centered for mobile

oliverguenther committed:
Remove bottom-margin in mobile, so that default modal styles apply

This fixes the datepicker banner from being hidden when there is not

enough height

oliverguenther committed:
Fix width to stay the same

oliverguenther committed:
Add pattern and inputmode to text-field component

This allows us to specify the virtual keyboard type

lindenthal commented:
Hi @oliverguenther and @psatyal,

I just had a quick preview on iOS FF:
- Is it possible to use the entire width?
- Do we need two close actions?

Can you please compare with Google Flights on mobile?

Best
Niels


![EAF7B73C-2B46-4601-B8ED-B5C46C303A75](https://user-images.githubusercontent.com/1019287/194213695-31580189-eab6-4125-b977-a93effe8308d.png)

oliverguenther commented:
@lindenthal, width of the datepicker is fixed by flatpickr. It doesn't support autoscaling, and implementing this ourselves is unfeasible.

The `X` close is shown for all mobile modals. We could hide the cancel button for mobile?

oliverguenther commented:
I extracted two bugs from this PR:

- The two cancel/close button issue: https://community.openproject.org/wp/44398
- Question regarding alignment: https://community.openproject.org/wp/44399

b12f commented on frontend/src/app/spot/components/text-field/text-field.component.ts:
```diff
@@ -46,6 +46,10 @@ export class SpotTextFieldComponent implements ControlValueAccessor {
 
+  @Input() public pattern:string|undefined;
+
+  @Input() public inputmode:string = 'text';
```
Could you add doc comments to both of these? If you merge `release/12.3`, you should see the others that are now there.

oliverguenther replied:
Done 👍

b12f reviewed (approved):
LGTM

cbliard committed:
Fix flickering frontend test

Shallow copy the registered tabs, so that modifying them in the test has
no impact on subsequent tests.

oliverguenther committed:
Input docs
